File: java/Abstract.java
package Abstract;

abstract class Animal {
    private String name;

    public Animal(String name) {
      this.name = name;
    }

    public String getName() {
      return name;
    }

    public abstract String getSound();
}

class Cat extends Animal {
    public Cat(String name) {
      super(name);
    }

    @Override
    public String getSound() {
      return "Miaw";
    }
}

public class Abstract {
    public static void main(String[] args) {
        Cat catus = new Cat("Mewo");
        System.out.println(catus.getName());
        System.out.println(catus.getSound());
    }
}

File: python/Abstract.py
from abc import ABC,abstractmethod
 
class Animal(ABC):
    @abstractmethod
    def getSound(self):
        pass

class Snake(Animal):
    def getSound(self):
        print("HISSS")

def main():
    obj = Snake()
    obj.getSound()

if __name__=="__main__":
    main()